parser/grammar.go

```diff
@@ -53,6 +53,106 @@ func (b *Bool) Capture(values []string) error {
 	return nil
 }
 
+type Block struct {
+	Comments []*Comment      `@Comment*`
+	Body     []*BlockElement `(@@ ";")*`
+}
+
+type BlockElement struct {
+	Check     *Check         `@@`
+	Predicate *Predicate     `|@@`
+	RuleBody  []*RuleElement `("<-" @@ ("," @@)*)?`
+}
+
+func (b *Block) ToBiscuit() (*biscuit.ParsedBlock, error) {
+	facts := []biscuit.Fact{}
+	rules := []biscuit.Rule{}
+	checks := []biscuit.Check{}
+	for _, e := range b.Body {
+		if e.Check != nil {
+			c, err := e.Check.ToBiscuit()
+			if err != nil {
+				return nil, err
+			}
+			checks = append(checks, *c)
+		} else if e.Predicate != nil && e.RuleBody != nil {
+			rule := Rule{
+				Head: e.Predicate,
+				Body: e.RuleBody,
+			}
+			r, err := rule.ToBiscuit()
+			if err != nil {
+				return nil, err
+			}
+			rules = append(rules, *r)
+		} else {
+			p, err := e.Predicate.ToBiscuit()
+			if err != nil {
+				return nil, err
+			}
+			facts = append(facts, biscuit.Fact{Predicate: *p})
+		}
+	}
+	return &biscuit.ParsedBlock{Facts: facts, Rules: rules, Checks: checks}, nil
+}
+
+type Authorizer struct {
+	Comments []*Comment           `@Comment*`
+	Body     []*AuthorizerElement `(@@ ";")*`
```

Comment:
how are there fancy tags used?

Reply:
these are used by [participle](https://github.com/alecthomas/participle)

The linter actually complains about those tags, since they are not qualified, but doing it would make the grammar way less readable.